src/render.rs

```diff
@@ -73,6 +71,10 @@ impl From<Source> for Resource {
     }
 }
 
+// TODO: I'm pretty sure you only need to implement for the reference.
+// once compiling, try taking it out and seeing what happens.
+// 
+// Also, like none of these fail. What are they being used with?
```

Comment:
@einargs The reason these are `TryFrom` instead of `From` is because they're not a strict 1:1 reversable relationship. The trait documentation states that if it's not a lossless conversion, it shouldn't be `From`, and should be `TryFrom` instead.

If we want to change them out for a straight `From` trait implementation, I'm okay with that. Just trying to be compliant to expectations.